src/metabase/models/setting.clj

```diff
@@ -384,14 +384,22 @@
   "Get the value, if any, of `setting-definition-or-name` from the DB (using / restoring the cache as needed)."
   ^String [setting-definition-or-name]
   (let [setting (resolve-setting setting-definition-or-name)]
-    (when (allows-site-wide-values? setting)
-      (let [v (if *disable-cache*
-                (db/select-one-field :value Setting :key (setting-name setting-definition-or-name))
-                (do
-                  (setting.cache/restore-cache-if-needed!)
-                  (clojure.core/get (setting.cache/cache) (setting-name setting-definition-or-name))))]
-        (when (seq v)
-          v)))))
+    ;; cannot use db (and cache populated from db) if db is not set up
+    (when-let [db-is-set-up? (resolve 'metabase.db/db-is-set-up?)]
```

Comment:
Should we be requiring `metabase.db` or at least using `requiring-resolve` here to make sure `metabase.db` is loaded? Not sure if it's happening somewhere else but it's probably best to be extra safe and put it there anyway so people don't see it and get worried that it's a potential race condition

Reply:
Sure it's a good thought. I've never understood the real difference between a built-in `requiring-resolve` and our own `classloader/require`. I think it is not relevant to things on the "regular" classpath and I can just use `requiring-resolve` in this case right?